### PR TITLE
DG Jump Error Scaling

### PR DESCRIPTION
**Pull request by pazner**
This PR adds some more options for scaling terms in `GridFunction::ComputeDGFaceJumpError`. For example, the natural norm in which to measure the error for DG diffusion scales the jumps by 1/h (or p^2/h), whereas for advection problems, the jump terms do not scale with h in the natural norm. This PR adds a class `JumpScaling` that allows the user to specify what type of scaling, and potentially combine with a constant factor (e.g. a penalty coefficient). The interface should remain backwards compatible.

A minor bug in `ConvergenceStudy` is also fixed, so that jump errors are reported even if H^1 errors are not.

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#1804](https://github.com/mfem/mfem/pull/1804) | @pazner | @v-dobrev | @psocratis + @vladotomov | 10/19/20 | 11/10/20 | 11/18/20 | |

### CODE REVIEW

**Comment by pazner** [comment]
Ping editors: @tzanio @v-dobrev :)

**Comment by v-dobrev** [comment]
> Ping editors: @tzanio @v-dobrev :)

Sorry for the delay, I did not realize it's been 11 days.

**Review by psocratis — APPROVED** [review]
Very nice and clean! 
Maybe adding 1-2 example runs in rates.cpp/prates.cpp demonstrating the changes? (i.e. the appropriate norm scalings) would be nice.

File: tests/convergence/rates.cpp [hunk 1]
```diff
@@ -26,6 +26,7 @@
 //               rates -m ../../data/inline-hex.mesh -sr 1 -prob 1 -o 2
 //               rates -m ../../data/square-disc.mesh -sr 2 -prob 1 -o 1
 //               rates -m ../../data/star.mesh -sr 2 -prob 3 -o 2
+//               rates -m ../../data/star.mesh -sr 2 -prob 3 -o 2 -j 0
```
Comment (vladotomov) [inline tests/convergence/rates.cpp:29]: Wasn't `-j 0` the wrong option for diffusion? Or are you trying to preserve the autotests? I think this should demonstrate the correct option.

**Review by vladotomov — APPROVED** [review]

File: tests/convergence/rates.cpp (hunk 1 above)
Comment (pazner) [reply tests/convergence/rates.cpp:29]: `-j 0` is not the "natural" norm, so the rates for the jump term will be different from those in the preceding sample run. It's not necessarily "wrong" though.

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Comment by v-dobrev** [comment]
LGTM

**Comment by v-dobrev** [comment]
Merged in `next` for testing...

**Comment by v-dobrev** [comment]
Re-merged in `next`.

**Comment by v-dobrev** [comment]
LGTM

**Pull request merged**